src/components/Document.tsx

```diff
@@ -0,0 +1,68 @@
+import { useEffect, useState } from "react";
+import Button from "./Button";
+import { User } from "../types/TableTypes";
+
+interface DocumentProps {
+  tenantId: string;
+}
+
+const Document: React.FC<DocumentProps> = ({ tenantId }) => {
+  const [tenant, setTenant] = useState<User | null>(null);
+  const [error, setError] = useState<string | null>(null);
+
+  useEffect(() => {
+    const fetchTenantDetailData = async () => {
+      try {
+        const response = await fetch("/assets/data.json");
+        if (!response.ok) {
+          throw new Error("Failed to fetch data");
+        }
+        const data: User[] = await response.json();
+        const selectedTenant = data.find((tenant) => tenant.id === tenantId);
+        if (selectedTenant) setTenant(selectedTenant);
+      } catch (err: any) {
+        setError(err.message || "An unknown error occurred");
+      }
+    };
+
+    fetchTenantDetailData();
+  }, []);
+
+  if (error) {
+    return <p className="text-red-500">Error: {error}</p>;
+  }
+
+  if (!tenant) {
+    return <p className="text-gray-500">Loading...</p>;
+  }
```

Comment:
It should be written only in the TenantDetailspage, no need to write it in every component